neqo-http3/src/connection_server.rs

```diff
@@ -49,12 +51,112 @@ impl Http3ServerHandler {
             .send_streams
             .get_mut(&stream_id)
             .ok_or(Error::InvalidStreamId)?
+            .http_stream()
+            .ok_or(Error::InvalidStreamId)?
             .set_message(headers, Some(data))?;
         self.base_handler
             .insert_streams_have_data_to_send(stream_id);
         Ok(())
     }
 
+    /// Accept a WebTransport Session request
+    pub(crate) fn wt_session_response(
+        &mut self,
+        conn: &mut Connection,
+        stream_id: u64,
+        accept: bool,
+    ) -> Res<()> {
+        let recv_stream = self.base_handler.recv_streams.get_mut(&stream_id);
+        if let Some(r) = &recv_stream {
+            if r.stream_type() != Http3StreamType::WebTransportSession {
+                return Err(Error::InvalidStreamId);
+            }
+        }
+
+        let send_stream = self.base_handler.send_streams.remove(&stream_id);
+        let recv_stream = self.base_handler.recv_streams.remove(&stream_id);
+        match (send_stream, recv_stream) {
+            (None, None) => Err(Error::InvalidStreamId),
+            (None, Some(_)) | (Some(_), None) => {
+                // TODO this needs a better error
+                self.base_handler.stream_reset(
+                    conn,
+                    stream_id,
+                    Error::HttpRequestRejected.code(),
+                )?;
+
+                Err(Error::InvalidStreamId)
+            }
+            (Some(mut s), Some(_)) => {
+                // TODO check that streams are not closed.
+                let headers = if accept {
+                    vec![Header::new(":status", "200")]
+                } else {
+                    vec![Header::new(":status", "400")]
+                };
+                s.http_stream()
+                    .ok_or(Error::InvalidStreamId)?
+                    .set_message(&headers[..], None)?;
+                WebTransportSession::create_session_server(
+                    stream_id,
+                    s,
+                    Box::new(self.events.clone()),
+                    &mut self.base_handler,
+                );
+                self.base_handler
+                    .insert_streams_have_data_to_send(stream_id);
+                Ok(())
+            }
+        }
+    }
+
+    /// Create a WebTraansport stream.
+    pub fn wt_create_stream(
+        &mut self,
+        conn: &mut Connection,
+        stream_id: u64,
+        stream_type: StreamType,
+    ) -> Res<u64> {
+        let wt_session = self
+            .base_handler
+            .send_streams
+            .get_mut(&stream_id)
+            .ok_or(Error::InvalidStreamId)?
+            .get_wt_session()
+            .ok_or(Error::InvalidStreamId)?;
+        match WebTransportSession::create_new_stream_local(wt_session, stream_type, conn)? {
```

Comment:
Note to self: this split probably could be improved.  We might need different handling entirely based on the stream type rather than have this awkward return value handling.